dpnp/dpnp_algo/dpnp_algo_arraycreation.pyx

```diff
@@ -262,34 +262,20 @@ cpdef dparray dpnp_tril(dparray m, int k):
     return result
 
 
-cpdef dparray dpnp_triu(m, k):
-    cdef dparray result
+cpdef dparray dpnp_triu(dparray m, int k):
     if m.ndim == 1:
+        res_shape=(m.shape[0], m.shape[0])
+    else:
+        res_shape=m.shape
 
-        result = dparray(shape=(m.shape[0], m.shape[0]), dtype=m.dtype)
-
-        for i in range(result.size):
-            ids = get_axis_indeces(i, result.shape)
-
-            diag_idx = max(-1, ids[result.ndim - 2] + k)
-            diag_idx = min(diag_idx, result.shape[result.ndim - 1])
+    cdef dparray result = dparray(shape=res_shape, dtype=m.dtype)
 
-            if ids[result.ndim - 1] >= diag_idx:
-                result[i] = m[ids[result.ndim - 1]]
-            else:
-                result[i] = 0
-    else:
-        result = dparray(shape=m.shape, dtype=m.dtype)
+    cdef DPNPFuncType param1_type = dpnp_dtype_to_DPNPFuncType(m.dtype)
 
-        for i in range(result.size):
-            ids = get_axis_indeces(i, result.shape)
+    cdef DPNPFuncData kernel_data = get_dpnp_function_ptr(DPNP_FN_TRIU, param1_type, param1_type)
 
-            diag_idx = max(-1, ids[result.ndim - 2] + k)
-            diag_idx = min(diag_idx, result.shape[result.ndim - 1])
+    cdef custom_1in_1out_func_ptr_t func = <custom_1in_1out_func_ptr_t > kernel_data.ptr
```

Comment:
this pointer type (`custom_1in_1out_func_ptr_t `) is used in several places on this src.
Bad name, should be renamed (not in this PR)